client-ui/src/app/components/dashboard/dashboard.component.ts

```diff
@@ -13,200 +13,131 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import {AfterViewInit, Component, ElementRef, OnDestroy, OnInit, ViewChild, ViewEncapsulation} from '@angular/core';
-import {Params} from "@angular/router";
+import {Component, OnDestroy, ViewEncapsulation} from '@angular/core';
 import {Store} from '@ngrx/store';
-import {DatatableComponent} from '@swimlane/ngx-datatable';
 import * as _ from 'lodash';
 import * as moment from 'moment';
-import {FormGroupState, NgrxValueConverter} from 'ngrx-forms';
-import {fromEvent, Observable, Subscription} from 'rxjs';
-import {debounceTime, filter, map, switchMap, tap} from 'rxjs/operators';
+import {Subscription} from 'rxjs';
+import {filter} from 'rxjs/operators';
 import {AppState} from 'src/app/model/app-state';
 import {RouterStateUrl} from 'src/app/model/routerStateUrl';
-import * as DashboardFormActions from 'src/app/store/actions/dashboard-form.actions';
+import * as DashboardActions from 'src/app/store/actions/dashboard.actions';
 import * as ExecutionEventsActions from 'src/app/store/actions/execution-events.actions';
 import * as RouterAction from 'src/app/store/actions/router.actions';
+import {DashboardVM} from "../../model/viewModels/dashboardVM";
+import {DateRange, Timestamp} from "../time-frame-picker/time-frame-picker.model";
+import {PageableExecutionEventsResponse} from "../../generated/models/pageable-execution-events-response";
+import {TableSort} from "./dashboard.model";
+import {DashboardLoadingIndicator} from "./dashboard.loading-indicator";
+import {ActivatedRoute} from "@angular/router";
+import {DashboardUrlStatePersister} from "./dashboard.url-state-persister";
+
+const SEARCH_TERM_UPDATE_DELAY = 300 //millis
 
 @Component({
   selector: 'dashboard',
   templateUrl: './dashboard.component.html',
   styleUrls: ['dashboard.component.less'],
   encapsulation: ViewEncapsulation.None
 })
-export class DashboardComponent implements OnInit, AfterViewInit, OnDestroy {
-  constructor(
-    private store: Store<AppState>
-  ) {
-    this.formState$ = store.select('dashboardForm')
-  }
-
-  @ViewChild('searchInput', { static: false }) searchInput: ElementRef
+export class DashboardComponent implements OnDestroy {
+  readonly asAtTime: Timestamp = +moment()
+  readonly loadingIndicator = new DashboardLoadingIndicator
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [expected member-variable-declaration: 'loadingIndicator' to have a typedef](https://app.codacy.com/app/ABSA_OSS/spline/pullRequest?prid=4491035)